docs/architecture.md

```diff
@@ -164,97 +171,194 @@ schedulingProfiles:
-of the keys are checked and the results are ANDed together.
-
-**DecodeFilter**<br>
-Filters out pods that are not marked either as decode or both prefill and decode. The filter
-looks for the label `llm-d.ai/role`, with a value of either `decode` or `both`. In addition pods
-that are missing the label will not be filtered out.<br>
-*Type:* decode-filter<br>
-*Parameters:* None<br>
-
-**PrefillFilter**<br>
-Filters out pods that are not marked as prefill. The filter looks for the label `llm-d.ai/role`,
-with a value of `prefill`.<br>
-*Type:* prefill-filter<br>
-*Parameters:* None<br>
-
-**PrefixCacheScorer**<br>
-The `prefix-cache-scorer` scores a request based on the KV cache localities.
-It supports two modes: `estimate` and `cache_tracking`.<br>
-
-**`estimate` mode** (default):<br>
-This mode uses the default GIE prefix scorer and scores pods based on how much of the prompt is estimated to be present in the pod’s KV cache.<br>
-*Type*: `prefix-cache-scorer`<br>
-*Parameters:*<br>
-
-\- `hashBlockSize`: Specifies the size of the blocks used to split the input **prompt** when calculating block hashes. Defaults to `64` if not specified.<br>
-\- `maxPrefixBlocksToMatch`: Specifies the maximum number of prefix blocks to match. Defaults to `256` if not specified.<br>
-\- `lruCapacityPerServer`: Specifies the capacity of the LRU indexer, in number of entries per server (pod). Defaults to `31,250` if not specified.<br>
-
-**Note:** \-  `mode: estimate` is not required, as it is the default.
-
-**`cache_tracking` mode**:<br>
-This mode scores requests based on the actual KV cache state in vLLM. It is more accurate than both `SessionAffinity` and `PrefixCachePlugin` in `estimate` mode,
-but incurs additional computation overhead to track the current cache state.<br>
-*Type*: `prefix-cache-scorer`<br>
-*Parameters:*<br>
-\- `mode: cache_tracking`<br>
-\- `kvCacheRedisAddr`: The address of the Redis instance used for cache tracking.
-Due to the sensitivity of this plugin’s parameters, the following environment variable is required when using `cache_tracking` mode:
-`HF_TOKEN`: The Hugging Face token to be used.
-
-**LoadAwareScorer**<br>
+#### PrefillHeader
+
+Sets a header for use in disaggregated prefill/decode
+
+- **Type**: `prefill-header-handler`
+- **Parameters**:
+  - `prefillProfile`: specifies the name of the profile used for the prefill scheduling. Only needed if the prefill profile is not named `prefill`.
+
+---
+
+#### PdProfileHandler
+
+Selects the profiles to use when running with disagregated prefill/decode
+
+- **Type**: `pd-profile-handler`
+- **Parameters**:
+  - `threshold`: specifies the threshold at which there are enough new input tokens to send the request to prefill and then decode, vs just to decode.
+  - `hashBlockSize`: specifies the length of the prompt chunk that a block is keyed by. This must the same value used for the PrefixCachePlugin.
```

Comment:
ok for now. for the long run it doesn't make sense to define same hashBlockSize in two places.
we might need to either add a parameters sections that all plugins can consume selected params, or change implementation of pd profile handler.

Reply:
Agreed. Again this is just a refactoring of the docs formatting, the extended parts are in the `prefix-cache-scorer` `cache_tracking` mode. Sorry for the confusion.